controllers/favorites.js

```diff
@@ -3,40 +3,49 @@ module.exports = (app) => {
   const RecipeSchema = require('../models/recipe');
   const UserSchema = require('../models/user');
 
-  // Show all recipes saved in their cart
-  app.get('/favorites', (req, res) => {
-    // find recipe, return ones that have been favorited 1+ times
-    RecipeSchema.find({
-      'usersWhoFavorited': {
-        $ne: null
-      }
-    }, function(err, recipes) {
-      if (err) {
-        console.error(err);
-      } else {
-        res.render('favorites', {
-          recipes: recipes
-        });
-      }
-    })
-  });
-
   // Send a POST request to the database to create the recipes collection
   app.post('/favorites/', (req, res) => {
     const favoriteId = req.body.favoriteId;
     const userId = app.locals.user.id
 
     // find recipe, add userId to usersWhoFavorited
-    RecipeSchema.findByIdAndUpdate(favoriteId, {
-      $addToSet: { usersWhoFavorited: userId }}, function(err) {
+    RecipeSchema.findById(favoriteId, function(err, favoriteInDB) {
       if (err) return handleError(err);
+      if (favoriteInDB.usersWhoFavorited.includes(userId)) {
+        // user already favorited, and is trying to un-favorite
+        RecipeSchema.findByIdAndUpdate(favoriteId, {
```

Comment:
also, `favoriteId` here is a bit confusing b/c we don't know what the favorite thing is. the thing we're talking about is a recipe so the better description is that this is a `recipeId` and we're talking about a `recipe` or `favoritedRecipe` if you wanted to be a bit more descriptive. to give you an idea...

```
app.post('/favorites/', (req, res) => {
  const recipeId = req.body.favoritedRecipeId;
  const userId = app.locals.user.id;

  // leslie comment - this is making it clear we are getting a recipe back in the cb
  RecipeSchema.findById(recipeId, function(err, recipe) {
    if (err) return handleError(err);
    if (recipe.favoritedUserIds.includes(userId)) {
      // user already favorited, and is trying to un-favorite
      RecipeSchema.findByIdAndUpdate(recipeId, {
        $pull: { favoritedUserIds: userId }}, function(err) {
          if (err) return handleError(err); 
      });
    }
    else {
      // user has not favorited before
      RecipeSchema.findByIdAndUpdate(recipeId, {
        $addToSet: { favoritedUserIds: userId }}, function(err) {
          if (err) return handleError(err);
        });
      }
    });
...
```